drivers/include/at.h

```diff
@@ -186,19 +209,28 @@ typedef struct {
 #endif
 } at_dev_t;
 
+/**
+ * @brief AT device initialization parameters
+*/
+typedef struct {
+    uart_t uart;            /**< UART device where the AT device is attached */
+    uint32_t baudrate;      /**< UART device baudrate */
+    char *rx_buf;           /**< UART rx buffer */
+    size_t rx_buf_size;     /**< UART rx buffer size */
+    char *err_buf;          /**< error buffer */
```

Comment:
Can this be `NULL`?

Reply:
no

Reply:
Would it make sense to allow this?
There was no error reporting before, so if a user doesn't need this, they might not want to pay for the additional buffer.

Reply:
I'll have a look at this next week, it makes sense.